crates/red_knot_python_semantic/src/types.rs

```diff
@@ -619,6 +651,136 @@ impl<'db> From<&Type<'db>> for Type<'db> {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum CallOutcome<'db> {
```

Comment:
I find the name `Outcome` a bit strange but I think that's something you plan to address as part of your context proposal.